### PR TITLE
delta: update to 0.10.2

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by ericonr** [comment]
```
   Compiling git-delta v0.10.2 (/builddir/delta-0.10.2)
error[E0277]: the trait bound `std::collections::HashSet<std::string::String>: From<[_; 0]>` is not satisfied
   --> src/utils/process.rs:689:46
    |
689 |             Some(CallingProcess::GitGrep(([].into(), [].into())))
    |                                              ^^^^ the trait `From<[_; 0]>` is not implemented for `std::collections::HashSet<std::string::String>`
    |
    = note: required because of the requirements on the impl of `Into<std::collections::HashSet<std::string::String>>` for `[_; 0]`

error[E0277]: the trait bound `std::collections::HashSet<std::string::String>: From<[_; 0]>` is not satisfied
   --> src/utils/process.rs:689:42
    |
689 |             Some(CallingProcess::GitGrep(([].into(), [].into())))
    |                                          ^^^^^^^^^^^^^^^^^^^^^^ the trait `From<[_; 0]>` is not implemented for `std::collections::HashSet<std::string::String>`
    |
    = note: required because of the requirements on the impl of `Into<std::collections::HashSet<std::string::String>>` for `[_; 0]`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `git-delta`

To learn more, run the command again with --verbose.
=> ERROR: delta-0.10.2_1: do_check: '${make_cmd} test --release --target ${RUST_TARGET} ${configure_args} 
```

tests fail to compile, do you think they are depending on new Rust features we don't have?

**Comment by cinerea0** [comment]
I swear, this happens every time I forget to run `xbps-src check` locally. Yes, this looks like a guarantee that could be proven by a newer version of the rust compiler. Would you prefer I close this or set to WIP?

**Comment by ericonr** [comment]
It's only check, IMO you can set `make_check=no` with an appropriate `# TODO/FIXME` comment

File: srcpkgs/delta/template [hunk 1]
```diff
@@ -9,7 +9,9 @@ maintainer="Andrew Benson <abenson+void@gmail.com>"
 license="MIT"
 homepage="https://github.com/dandavison/delta"
 distfiles="${homepage}/archive/${version}.tar.gz"
-checksum=f002a94119cd5b6fd9deede7f344667358baf5015d0051e055fec6334ee0653f
+checksum=e0f71d72eca543478941401bd96fefc5fa3f70e7860a9f858f63bfecf8fd77a5
+# TODO: Re-enable checks when Rust gets updated
+make_check=no
```
Comment (ericonr) [inline srcpkgs/delta/template:14]:
```suggestion
make_check=no # TODO: Re-enable checks when Rust gets updated
```

So `grep make_check=no` gives you the justification too :)

**Comment by ericonr** [comment]
Leave a link so I see the mention on my PR #32555

**Pull request merged**